Automattic/jetpack
# Update/Use the new Terms of Service package in Jetpack

This PR deprecated the Jetpack ::jetpack_tos_agreed().
Updates any calls that relied on that method the the new Terms_of_service->has_agreed();

Also instantiates any Jetpack tracking as soon as the sites agrees to the TOS.

Supersedes #13759

#### Changes proposed in this Pull Request:
* Fixes tracking of new connections, integrates the new Terms Of Service Package in Jetpack.


#### Testing instructions:
* Connect Jetpack, does it work as expected. 
* Does tracking for the connected user work as expected?
* Do you see the jetpack_connection_success again? After the site is connected.

#### Proposed changelog entry for your changes:
* Integrates the new Terms Of Service Package in Jetpack
